src/scaffoldfitter/fitter.py

```diff
@@ -32,19 +33,28 @@ def _next_available_identifier(node_set, candidate):
 
 class Fitter:
 
-    def __init__(self, zincModelFileName: str, zincDataFileName: str):
+    def __init__(self, zincModelFileName: str=None, zincDataFileName: str=None, region: Region=None):
         """
-        :param zincModelFileName: Name of zinc file supplying model to fit.
-        :param zincDataFileName: Name of zinc filed supplying data to fit to.
+        Create instance of Fitter either from model and data file names, or the model/fit region.
+        :param zincModelFileName: Name of zinc file supplying model to fit, or None if supplying region.
+        :param zincDataFileName: Name of zinc filed supplying data to fit to, or None if supplying region.
+        :param region: Region in which to build model and perform fitting, or None if supplying file names.
         """
         self._zincModelFileName = zincModelFileName
         self._zincDataFileName = zincDataFileName
-        self._context = Context("Scaffoldfitter")
+        if region:
+            assert (zincModelFileName is None) and (zincDataFileName is None)
```

Comment:
If you are favouring regions over names then shouldn't you just set these things `None` here and document the preference.

Reply:
I'm keeping both approaches going, and not breaking the current code using filenames.